homeassistant/components/denonavr/media_player.py

```diff
@@ -54,8 +54,44 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_SOUND_MODE_RAW = "sound_mode_raw"
+ATTR_BASS = "bass"
+ATTR_BASS_LEVEL = "bass_level"
 ATTR_DYNAMIC_EQ = "dynamic_eq"
+ATTR_DYNAMIC_VOLUME = "dynamic_volume"
+ATTR_MULTI_EQ = "multi_eq"
+ATTR_RECEIVER_TYPE = "receiver_type"
+ATTR_REFERENCE_LEVEL_OFFSET = "reference_level_offset"
+ATTR_SOUND_MODE_RAW = "sound_mode_raw"
+ATTR_SUPPORT_SOUND_MODE = "support_sound_mode"
+ATTR_TELNET_CONNECTED = "telnet_connected"
+ATTR_TELNET_HEALTHY = "telnet_healthy"
+ATTR_TONE_CONTROL_ADJUST = "tone_control_adjust"
+ATTR_TONE_CONTROL_STATUS = "tone_control_status"
+ATTR_TREBLE = "treble"
+ATTR_TREBLE_LEVEL = "treble_level"
+ATTR_ZONE = "zone"
+
+EXTRA_STATE_ATTRIBUTES = {
+    ATTR_RECEIVER_TYPE,
+    ATTR_SUPPORT_SOUND_MODE,
+    ATTR_TELNET_CONNECTED,
+    ATTR_TELNET_HEALTHY,
+    ATTR_ZONE,
+}
+
+EXTRA_STATE_ATTRIBUTES_ON = {
```

Comment:
I think all these extra attributes need to be added as sensor entities instead of attributes.
I don't think this is allowed, but I will ask for a second opinion.

The benefit of sensors would be you get a history, users can enable/disable the ones they want and it is better for the state_machine/database.

Reply:
Agreed!